core/src/test/java/kafka/log/remote/RemoteLogManagerTest.java

```diff
@@ -558,6 +618,38 @@ void testCopyLogSegmentsToRemoteShouldNotCopySegmentForFollower() throws Excepti
         verify(mockLog, never()).updateHighestOffsetInRemoteStorage(anyLong());
     }
 
+    @Test
+    void testCopyLogSegmentsToRemoteShouldNotCopySegmentWithCorruptedIndexes() throws Exception {
```

Comment:
I might be missing something very obvious but where are we corrupting the indexes in this test?

Reply:
I do not declare timeindex, 
```
when(oldSegment.timeIndex()).thenReturn(new TimeIndex(TestUtils.tempFile(), baseOffset, maxEntries * 12));
```

what means this file is missing

Reply:
can we add test for other types of corruption as well? e.g. in the disk full issue that you mentioned in the description, the "corruption" is not index missing, instead it is an existing index file not having all entries as expected.